torchmetrics/collections.py

```diff
@@ -185,6 +147,51 @@ def persistent(self, mode: bool = True) -> None:
         for _, m in self.items():
             m.persistent(mode)
 
+    def add_metrics(self, metrics: Union[Metric, Sequence[Metric], Dict[str, Metric]],
+                    *additional_metrics: Metric) -> None:
+        """Add new metrics to Metric Collection
+        """
+        if isinstance(metrics, Metric):
+            # set compatible with original type expectations
+            metrics = [metrics]
+        if isinstance(metrics, Sequence):
```

Comment:
I understand this is just copying what we had here before, but we should probably have `Iterable` here instead of `Sequence` as we doin't really care about indexing.

Reply:
but iterable does not have len, right?

Reply:
We are not using len though, we are just directly converting to a list in the next line.